tests/unit/lms/models/application_settings_test.py

```diff
@@ -43,6 +44,17 @@ def test_set(self, application_settings, group, key, value, expected_value):
 
         assert application_settings[group][key] == expected_value
 
+    def test_secrets(self, application_settings, aes_service):
+        application_settings.set_secret(aes_service, "GROUP", "KEY", "VERY SECRET")
+
+        assert (
+            application_settings.get_secret(aes_service, "GROUP", "KEY")
+            == "VERY SECRET"
+        )
```

Comment:
Do we want to make any assertions to ensure we called the AES service and didn't store the content in the clear? I think this would pass if it didn't encrypt anything?

I think a decent pattern for this is to have three tests:

 * One which sets a value and compares the results with baked values (you might have to do something to fix the IV?)
 * One with baked values to see it decodes
 * One round trip like this

Of these the last two probably prove most of what we want and are easy.

This has the nice effect of protecting you from arbitrary changes which still round trip, but make stored values unable to be decrypted. On the other hand perhaps we've proved the AES service works and we need to ensure we call it correctly?

So I guess, either we should lean into mockist tests and mock the service, or do more integrated tests and use real values.